PCF-ADR-0002-release-story-and-branch-maintenance.md

```diff
@@ -0,0 +1,140 @@
+steps for generating build artifacts, etc..
+
+We do not need to create major version tags unless there is a significant change in the software architecture or
+functionality that requires a major version bump.
+
+### Versioning Scheme
+
+All projects should use semantic versioning scheme as it gives the most warning for potentially breaking changes when
+compared to other schemes such as calendar versioning.
+
+## Non Goals
+
+- Dictating how to build code for release (prefer automation, but a manual process is also okay)
+- Setting a release cycle for the codeflare project.
+
+## Open questions
+
+- What artifacts should be built to go alongside the releases? For example, should we generate documentation, binaries,
+  packages, etc.?
+- How should we communicate our releases to our stakeholders and customers? Should we use changelogs, release notes,
+  newsletters, etc.?
+
+## Alternatives
+
+### Git Flow
+
+GitFlow is no longer a popular branching and release strategy as it requires more overhead with branch management and
+more involved merging. One benefit that it does offers is a slower dev process for junior devs.
+
+### Trunk Based Development (TBD)
+
+TBD and GitHub flow are very similar in that they treat the main branch as a deployable branch that all feature branches
+and bug fixes should start from. The main difference comes with expectations on how often developers pull changes into
+the repo. TBD discourages branches living longer than a couple of days, any features that are unfinished or untested are
+gated using feature flags. GitHub Flow encourages thorough testing before merging anything into the main branch. TBD is
+difficult to implement in an open source community where many developers don't have direct access to the main branch, or
+an easy way to get one on one time with a maintainer to do a code review.
+
+## Risks
+
+GitHub Flow and Trunk Based Development both require discipline keeping tests up to date and ensuring that the `main`
+branch is always in a deployable state.
+
+## Stakeholder Impacts
+
+| Group                         | Key Contacts     | Date       | Impacted? |
+| ----------------------------- | ---------------- | ---------- | --------- |
+| CodeFlare SDK                 | key contact name | 4/19/2023  | yes       |
+| MCAD                          | key contact name | 4/19/2023  | yes       |
+| InstaScale                    | key contact name | 4/19/2023  | yes       |
+| CodeFlare Operator            | key contact name | 4/19/2023  | yes       |
+
+## References
+
+* https://docs.github.com/en/get-started/quickstart/github-flow
+* https://trunkbaseddevelopment.com/
+* https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow
+
+## Reviews
+
+| Reviewed by                   | Date       | Notes |
+| ----------------------------- | ---------  | ------|
+| name                          | date       | ?     |
```

Comment:
We should add at least Alex, myself, Mustafa, and Abhishek to this list of reviewers. Additional people are welcome to review.